src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/components/options_list_control.tsx

```diff
@@ -193,38 +218,39 @@ export const OptionsListControl = ({
   ]);
 
   const button = (
-    <EuiFormControlButton
-      role="combobox"
-      compressed={isCompressed(componentApi)}
-      iconType={loading ? 'empty' : 'arrowDown'}
-      iconSide="right"
-      value={hasSelections || existsSelected ? selectionDisplayNode : ''}
-      placeholder={displaySettings.placeholder ?? OptionsListStrings.control.getPlaceholder()}
+    <EuiFilterButton
+      badgeColor="success"
+      isLoading={loading}
+      iconType={'arrowDown'}
+      data-test-subj={`optionsList-control-${componentApi.uuid}`}
       css={styles.filterButton}
       onClick={() => setPopoverOpen(!isPopoverOpen)}
+      isSelected={isPopoverOpen}
+      numActiveFilters={selectedOptionsCount}
+      hasActiveFilters={Boolean(selectedOptionsCount)}
+      textProps={{ css: styles.filterButtonText }}
       aria-label={panelTitle ?? defaultPanelTitle}
       aria-expanded={isPopoverOpen}
       aria-controls={popoverId}
-      data-test-subj={`optionsList-control-${componentApi.uuid}`}
+      role="combobox"
     >
-      {Boolean(selectedOptionsCount) && (
-        <EuiNotificationBadge color="success">{selectedOptionsCount}</EuiNotificationBadge>
-      )}
-    </EuiFormControlButton>
+      {hasSelections || existsSelected
+        ? selectionDisplayNode
+        : displaySettings.placeholder ?? OptionsListStrings.control.getPlaceholder()}
+    </EuiFilterButton>
```

Comment:
We had to switch back to `EuiFilterButton` (undoing the changes in https://github.com/elastic/kibana/issues/237661) in order to get the loading state to work for both control panels and pinned controls.